src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -202,6 +205,23 @@ internal DecompressionMethods AutomaticDecompression
             }
         }
 
+        internal bool PreAuthenticate
+        {
+            get
+            {
+                return _preAuthenticate;
+            }
+            set
+            {
+                CheckDisposedOrStarted();
+                _preAuthenticate = value;
+                if (value)
+                {
+                    _credentialCache = new CredentialCache();
+                }
+            }
+        }
```

Comment:
About the race-condition alone, the documentation makes a statement that the "Instance members of HttpClientHandler are not guaranteed to be thread-safe". But as for the general issue of property setting to be completed before starting an operation, am afraid I did not find any documentation.